apis/ec2/v1alpha1/custom_types.go

```diff
@@ -60,28 +60,103 @@ type CustomVolumeParameters struct {
 	// The ID of the VPC with which you are creating the VPC peering connection.
 	// You must specify this parameter in the request.
-	PeerVPCID *string `json:"peerVPCID,omitempty"`
+	PeerVPCID *string `json:"peerVpcId,omitempty"`
 	// PeerVPCIDRef is a reference to an API used to set
 	// the PeerVPCID.
 	// +optional
-	PeerVPCIDRef *xpv1.Reference `json:"peerVPCIDRef,omitempty"`
+	PeerVPCIDRef *xpv1.Reference `json:"peerVpcIdRef,omitempty"`
 	// PeerVPCIDSelector selects references to API used
 	// to set the PeerVPCID.
 	// +optional
-	PeerVPCIDSelector *xpv1.Selector `json:"peerVPCIDSelector,omitempty"`
+	PeerVPCIDSelector *xpv1.Selector `json:"peerVpcIdSelector,omitempty"`
 	// Automatically accepts the peering connection. If this is not set, the peering connection
 	// will be created, but will be in pending-acceptance state. This will only lead to an active
 	// connection if both VPCs are in the same tenant.
 	AcceptRequest bool `json:"acceptRequest,omitempty"`
+
+	// Metadata tagging key value pairs
+	// +optional
+	Tags []Tag `json:"tags,omitempty"`
+}
+
+// CustomTransitGatewayParameters are custom parameters for TransitGateway
+type CustomTransitGatewayParameters struct {
+	// Metadata tagging key value pairs
+	// +optional
+	Tags []Tag `json:"tags,omitempty"`
+}
+
+// CustomTransitGatewayVPCAttachmentParameters are custom parameters for TransitGatewayVPCAttachment
+type CustomTransitGatewayVPCAttachmentParameters struct {
+	// The ID of the VPC.
+	// +optional
+	// +crossplane:generate:reference:type=github.com/crossplane/provider-aws/apis/ec2/v1beta1.VPC
+	// +crossplane:generate:reference:refFieldName=VPCIDRef
+	// +crossplane:generate:reference:selectorFieldName=VPCIDSelector
+	VPCID *string `json:"vpcId,omitempty"`
+
+	// VPCIDRef is a reference to an API used to set
+	// the VPCID.
+	// +optional
+	VPCIDRef *xpv1.Reference `json:"vpcIdRef,omitempty"`
+
+	// VPCIDSelector selects references to API used
+	// to set the VPCID.
+	// +optional
+	VPCIDSelector *xpv1.Selector `json:"vpcIdSelector,omitempty"`
+
+	// The IDs of one or more subnets. You can specify only one subnet per Availability
+	// Zone. You must specify at least one subnet, but we recommend that you specify
+	// two subnets for better availability. The transit gateway uses one IP address
+	// from each specified subnet.
+	// +optional
+	// +crossplane:generate:reference:type=github.com/crossplane/provider-aws/apis/ec2/v1beta1.Subnet
+	// +crossplane:generate:reference:refFieldName=SubnetIDsRefs
+	// +crossplane:generate:reference:selectorFieldName=SubnetIDsSelector
+	SubnetIDs []*string `json:"subnetIds,omitempty"`
+
+	// SubnetIDsRefs is a list of references to SubnetIDs used to set
+	// the SubnetIDs.
+	// +optional
+	SubnetIDsRefs []xpv1.Reference `json:"subnetIdsRefs,omitempty"`
+
+	// SubnetIDsSelector selects references to SubnetIDs used
+	// to set the SubnetIDs.
+	// +optional
+	SubnetIDsSelector *xpv1.Selector `json:"subnetIdsSelector,omitempty"`
```

Comment:
```suggestion
	// +crossplane:generate:reference:refFieldName=SubnetIDRefs
	// +crossplane:generate:reference:selectorFieldName=SubnetIDSelector
	SubnetIDs []*string `json:"subnetIds,omitempty"`

	// SubnetIDsRefs is a list of references to SubnetIDs used to set
	// the SubnetIDs.
	// +optional
	SubnetIDRefs []xpv1.Reference `json:"subnetIdRefs,omitempty"`

	// SubnetIDSelector selects references to SubnetIDs used
	// to set the SubnetIDs.
	// +optional
	SubnetIDSelector *xpv1.Selector `json:"subnetIdSelector,omitempty"`
```

We usually go for plural in `Refs` instead of the field in case of arrays.